recipes/mariadb-connector-c/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+        if self.settings.os != "Windows" and self.options.with_ssl == "schannel":
+            raise ConanInvalidConfiguration("schannel only supported on Windows")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        if self.options.get_safe("with_iconv"):
+            self.requires("libiconv/1.16")
+        if self.options.with_curl:
+            self.requires("libcurl/7.73.0")
+        if self.options.with_ssl == "openssl":
+            self.requires("openssl/1.1.1h")
+        elif self.options.with_ssl == "gnutls":
+            raise ConanInvalidConfiguration("gnutls not yet available in CCI")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "mariadb-connector-c-{}-src".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "${ZLIB_LIBRARY}",
+                              "${ZLIB_LIBRARIES}")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "IF(OPENSSL_FOUND)",
+                              "IF(OpenSSL_FOUND)")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})",
+                              "SET(SSL_LIBRARIES ${OpenSSL_LIBRARIES})")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "${OPENSSL_INCLUDE_DIR}",
+                              "${OpenSSL_INCLUDE_DIR}")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.os == "Windows":
+            self._cmake.definitions["WITH_MSI"] = False
+            self._cmake.definitions["WITH_SIGNCODE"] = False
+            self._cmake.definitions["WITH_RTC"] = False
+        else:
+            self._cmake.definitions["WITH_MYSQLCOMPAT"] = False
+            self._cmake.definitions["WITH_ICONV"] = self.options.with_iconv
+        self._cmake.definitions["WITH_UNIT_TESTS"] = False
+        self._cmake.definitions["WITH_DYNCOL"] = self.options.dyncol
+        self._cmake.definitions["WITH_EXTERNAL_ZLIB"] = True
+        self._cmake.definitions["WITH_CURL"] = self.options.with_curl
+        self._cmake.definitions["WITH_SSL"] = self.options.with_ssl
+        self._cmake.definitions["INSTALL_BINDIR"] = os.path.join(self.package_folder, "bin")
+        self._cmake.definitions["INSTALL_LIBDIR"] = os.path.join(self.package_folder, "lib")
+        self._cmake.definitions["INSTALL_PLUGINDIR"] = os.path.join(self.package_folder, "lib", "plugin")
```

Comment:
I think the MSVC error is caused by the backspace escaping characters.
This means that some build path might get embedded in the executable (e.g. the plugins).
So this library maybe requires some environment variables set.

```suggestion
        self._cmake.definitions["INSTALL_BINDIR"] = os.path.join(self.package_folder, "bin").replace("\\", "/")
        self._cmake.definitions["INSTALL_LIBDIR"] = os.path.join(self.package_folder, "lib").replace("\\", "/")
        self._cmake.definitions["INSTALL_PLUGINDIR"] = os.path.join(self.package_folder, "lib", "plugin").replace("\\", "/")
```

Reply:
On the other hand, it fails with Visual Studio 16 only (on my computer it doesn't fail, but long path is enabled at system level). Maybe just add `short_paths`?
Regarding environment variables, yes maybe, need to dig a little bit more.